webapi/src/main/scala/org/knora/webapi/messages/v1/responder/usermessages/UserMessagesV1.scala

```diff
@@ -39,59 +83,228 @@ sealed trait UsersResponderRequestV1 extends KnoraRequestV1
+  * @param apiRequestID the ID of the API request.
+  */
+case class UserUpdateRequestV1(userIri: webapi.IRI,
+                               propertyIri: webapi.IRI,
+                               newValue: Any,
+                               userProfile: UserProfileV1,
+                               apiRequestID: UUID) extends UsersResponderRequestV1
+
+
+// Responses
+
+/**
+  * Represents an answer to an user creating/modifying operation.
+  *
+  * @param userProfile the new user profile of the created/modified user.
+  * @param userData    information about the user that made the request.
+  */
+case class UserOperationResponseV1(userProfile: UserProfileV1, userData: UserDataV1) extends KnoraResponseV1 {
+    def toJsValue = UserV1JsonProtocol.userOperationResponseV1Format.write(this)
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Components of messages
+
+/**
   * Represents a user's profile.
   *
-  * @param userData     basic information about the user.
-  * @param groups       the groups that the user belongs to.
-  * @param projects     the projects that the user belongs to.
-  * @param isSystemUser `true` if this [[UserProfileV1]] represents the Knora API server itself.
+  * @param userData basic information about the user.
+  * @param groups   the groups that the user belongs to.
+  * @param projects the projects that the user belongs to.
+  * @param sessionId the sessionId,.
+  * @param permissionData the user's permission data.
   */
-case class UserProfileV1(userData: UserDataV1 = UserDataV1(lang = "en"), groups: Seq[IRI] = Nil, projects: Seq[IRI] = Nil, isSystemUser: Boolean = false) {
+case class UserProfileV1(userData: UserDataV1 = UserDataV1(lang = "en"),
+                         groups: Seq[IRI] = Vector.empty[IRI],
+                         projects: Seq[IRI] = Vector.empty[IRI],
+                         sessionId: Option[String] = None,
+                         isSystemUser: Boolean = false,
+                         permissionData: PermissionDataV1 = PermissionDataV1()
+                        ) {
+
+    /**
+      * Check password using either SHA-1 or BCrypt. The BCrypt password always starts with '$2a$'
+      *
+      * @param password the password to check.
+      * @return true if password matches and false if password doesn't match.
+      */
     def passwordMatch(password: String): Boolean = {
+        userData.password.exists {
+            hashedpassword => hashedpassword match {
+                case hp if hp.startsWith("$2a$") => {
+                    println(s"UserProfileV1 - passwordMatch - password: $password, hashedpassword: $hashedpassword")
+                    import org.mindrot.jbcrypt.BCrypt
+                    BCrypt.checkpw(password, hp)
+                }
+                case hp => {
+                    val md = java.security.MessageDigest.getInstance("SHA-1")
```

Comment:
This is only here because ``SHA-1`` was used in the "old" Salsah. In Knora we use ``BCrypt``.